File: examples/_1_PubSub/subscriber.cpp
#include <iostream>
#include <thread>
#include "SharedPubSub.hpp"
using namespace std;

int main(){
    shps::Subscriber<int> subscriber("PubSub", "PubSubSubscriber1",true);
    optional<int> value = nullopt;
    while(1){
        /*--------- Example 1 : Wait for notification ---------*/

        value = subscriber.readWait();
        // Verify if the queue had a value on notification.
        if(value.has_value()){
            cout << "SUBSCRIBER : " << dec << value.value() << endl;
        }
        else{
            cout << "SUBSCRIBER : No value in queue" << endl;
        }

        /*--------- Example 2 : Wait for notification with timeout ---------*/
        value = subscriber.readWait(2s);
        // Verify if the queue had a value on notification.
        if(value.has_value()){
            cout << "SUBSCRIBER : " << dec << value.value() << endl;
        }
        else{
            cout << "SUBSCRIBER : No value in queue" << endl;
        }
    }
    return 0;
}

File: examples/python/_1_PubSub/subscriber.py
from SharedPubSub import *
## Watchout :
# If readWait() without timeout is in the main thread,
# Use ctrl+\ instead of ctrl+c to exit.

subscriber = Subscriber_int("PubSub","PubSubSubscriberPy",True)

while(True):
    ### Example 1 : Wait for notification ###
    value = subscriber.readWait()
    print("SUBSCRIBER PY:",value if value!=None else "No value in queue")

    ### Example 2 : Wait for notification with a timeout ###
    value = subscriber.readWaitMS(2000) # in milliseconds
    print("SUBSCRIBER PY:",value if value!=None else "No value in queue")